plugins/scaffolder-backend-module-openchoreo/src/actions/component.ts

```diff
@@ -229,8 +231,21 @@ export const createComponentAction = (
 
         // Create the API client using the auto-generated client
         const baseUrl = config.getString('openchoreo.baseUrl');
+
+        // Get service token for authentication
+        let token: string | undefined;
+        if (tokenService.hasServiceCredentials()) {
+          try {
+            token = await tokenService.getServiceToken();
+            ctx.logger.debug('Using service token for OpenChoreo API');
+          } catch (error) {
+            ctx.logger.warn(`Failed to get service token: ${error}`);
+          }
+        }
+
         const client = createOpenChoreoApiClient({
```

Comment:
Shouldn't we use the user token instead of service token for resource management operations?

Reply:
Found this discussion https://github.com/backstage/backstage/issues/5214 and https://backstage.io/docs/features/software-templates/experimental#form-decorators experimental feature.
 
 With these two it seems we can pass the user token as a secret. 
 
  User fills form → Form Decorator runs → Gets user's x-openchoreo-token →
  Injects as OPENCHOREO_USER_TOKEN secret → Scaffolder action reads ctx.secrets →
  Uses user token for OpenChoreo API call → OpenChoreo can do user-based authz

Reply:
@mevan-karu I will explore this path and fix for the next release